### PR TITLE
Cleanup login / dev

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Normalize dev store. Make immutable
- [x] Normalize login store (not immutable yet)
- [x] Fix flow cycle

### CODE REVIEW

File: shared/actions/login.js [hunk 1]
```diff
@@ -33,16 +33,13 @@ const InputCancelError = {
   desc: 'Cancel Login',
 }
 
-const codePageSelector = ({login: {codePage}}: TypedState) => codePage
```
Comment (chrisnojima) [inline shared/actions/login.js]: get to get rid of this nesting now 🔥

File: shared/actions/tracker.js [hunk 2]
```diff
@@ -563,7 +563,7 @@ function _userChanged(action: {payload: {uid: string}}, state: TypedState) {
   if (username) {
     actions.push(TrackerGen.createGetProfile({username}))
   }
-  return actions
+  return Saga.all(actions)
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: fix saga warning about arrays

File: shared/constants/dev.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+import * as I from 'immutable'
+import type {_State} from './types/dev'
 
-export const initialState: State = {
-  debugConfig: {
-    dumbFilter: '',
-    dumbFullscreen: false,
-    dumbIndex: 0,
-  },
+const makeState: I.RecordFactory<_State> = I.Record({
```
Comment (chrisnojima) [inline shared/constants/dev.js]: immutable and get rid of hierarchy

File: shared/constants/login.js [hunk 4]
```diff
@@ -52,24 +52,17 @@ export function qrGenerate(code: string): string {
 }
 
 export const initialState: Types.State = {
-  codePage: {
```
Comment (chrisnojima) [inline shared/constants/login.js]: pull this down a level

File: shared/engine/index.js.flow [hunk 5]
```diff
@@ -1,8 +1,6 @@
 // @flow
 // Sep. engine file so we can avoid import cycles w/ flow-types.js
 // A lot of weak typing here as we're usually interfacing with this through helpers, could make it more accurate later
-import type {TypedState} from '../constants/reducer'
```
Comment (chrisnojima) [inline shared/engine/index.js.flow]: this made flow very unhappy

**Comment by chrisnojima** [comment]
@keybase/react-hackers i was going to make some more things immutable but had to normalize some things first

**Comment by chrisnojima** [comment]
@keybase/react-hackers ptal, i merged down a branch from this which marco approved  (https://github.com/keybase/client/pull/9753/commits/86646148df85ca52eae4047e6d9a484102fd229e) so you can skip that part

File: shared/constants/types/folders.js [hunk 6]
```diff
@@ -1,9 +1,31 @@
...
+  theme: 'public' | 'private',
+  size?: 'Small' | 'Large', // defaults to Large
+  name: string,
+  path: string,
+  lastModifiedMeta?: string,
+  lastModifiedBySelf?: ?boolean,
+  lastModifiedBy?: ?string,
+  modifiedMarker: boolean,
+  fileIcon: IconType,
+  onClick: () => void,
```
Comment (MarcoPolo) [inline shared/constants/types/folders.js]: Should this have an `onClick` here? I remember having an issue with having data for a component vs data that we store.

Comment (MarcoPolo) [reply shared/constants/types/folders.js]: Also did you delete the other FileProps definition?

Comment (chrisnojima) [reply shared/constants/types/folders.js]: yeah we don't want onclick in the store ever. this whole files section is (thankfully) going away soon. This pr just mostly moves stuff around so how it works is largely the same

**Pull request merged**